include/linux/double-fetch-detection.h

```diff
@@ -0,0 +1,17 @@
+#ifndef __LINUX_DOUBLE_FETCH_DETECTION_H__
+#define __LINUX_DOUBLE_FETCH_DETECTION_H__
+#ifdef CONFIG_DF_DETECTION
+
+struct address{
+    void *startaddress;
```

Comment:
Nit: I'd probably suggest `start_address`, `start_addr`, `start`, or `addr` for this one. Your call.